packages/mural-client/src/fetch.ts

```diff
@@ -47,61 +48,70 @@ let fetchConfig: AuthenticatedFetchConfig;
 
 export const authenticated = () => {
   const session = fetchConfig.sessionStore.get();
-  return !!(session && !isTokenExpired(session.refreshToken));
+  return !!(session && !isTokenExpiredOrWillSoon(session.refreshToken));
 };
 
 const authenticatedFetch = async (
   input: RequestInfo,
   init: RequestInit = {},
 ): Promise<Response> => {
-  // setup all the handlers
-  await verifyTokensExpiration();
-
+  const session = await getSession();
   const initWithAuth = {
     ...init,
-    headers: withAuthenticationToken(init.headers || {}),
+    headers: {
+      ...init.headers,
+      ...(session ? { Authorization: `Bearer ${session?.accessToken}` } : {}),
+    },
   };
 
   try {
     const response = await fetch(input, initWithAuth);
     return await checkStatus(response);
   } catch (err: any) {
-    const errorHandler = catchAuthenticationError(input, initWithAuth);
+    const errorHandler = catchAuthenticationError();
     return errorHandler(err);
   }
 };
 
-function withAuthenticationToken(headers: HeadersInit) {
+const getSession = (): Promise<Session | null> => {
   const session = fetchConfig.sessionStore.get();
```

Comment:
**nitpick**: This could be async instead to remove those `Promise.resolve(null)`.

```
const getSession = async (): Promise<Session | null> => {
  const session = fetchConfig.sessionStore.get();
  if (!session) return null;

  if (isTokenExpiredOrWillSoon(session.refreshToken)) {
    fetchConfig.sessionStore.delete();
    return null;
  }

  return handleTokenRefresh(session);
}
```